std/testing/asserts_test.ts

```diff
@@ -409,3 +409,24 @@ Deno.test({
     );
   },
 });
+
+Deno.test({
+  name: "failed with string contains quote",
+  fn(): void {
+    assertThrows(
+      (): void =>
+        assertEquals(
+          'John said, "Hello!" to Dave.',
+          'John said, "Hello" to Dave.'
+        ),
+      AssertionError,
+      [
+        "Values are not equal:",
+        ...createHeader(),
+        removed(`-   "John said, "Hello!" to Dave."`),
```

Comment:
This doesn't seem correct to me. There are unescaped double quotes nested inside double quotes in this string.

Reply:
It doesn't generate a failed assertion as such, but if you look at the issue I raised #5590 it outputs escaping slashes for no benefit. This could lead to confusion when running tests where the inputted string to the test is unknown. The error message should be representative of what was inputted.

Reply:
@RobDWaller The escapes seem unnecessary when you use the message programmatically and write it like `'John said, "Hello!" to Dave.'`. You shouldn't be testing a prettified string like this. The purpose of `format()` and assertion error messages are pretty printing to the terminal. A prettified string has quotes put around it and so the quotes inside of it are escaped.

> It doesn't generate a failed assertion as such

I meant for you to have an assertion fail on purpose, and tell me what's wrong with the way the output is displayed in the terminal. I'll do it:
```js
import { assertEquals } from "./std/testing/asserts.ts";

assertEquals(`John said, "Hello!" to Dave.`, "");
```
Before:

<kbd>![image](https://user-images.githubusercontent.com/29990554/85520277-984f3280-b5fa-11ea-97c9-a8ff0680e345.png)</kbd>

After:

<kbd>![image](https://user-images.githubusercontent.com/29990554/85520088-5a520e80-b5fa-11ea-98c9-f2be816ec986.png)</kbd>

The first one seems correct. Your patch takes away the backslashes before the nested double quotes, which would be incorrect.

Reply:
@nayeemrmn the purpose of the issue I raised and this pull request was really to find out if this was expected behavior as it struck me as confusing.

Given this is expected behavior then I'm happy to close this pull request and the issue #5590. 

Maybe I will write some additional documentation on asserting error messages so this behavior is clear for all users of Deno.